lms/djangoapps/labster_course_license/views.py

```diff
@@ -58,6 +61,45 @@ def license_handler(request, course, ccx=None):
     return HttpResponseBadRequest('Only GET and POST methods are supported.')
 
 
+class LicensedSimulationsUpdateView(APIView):
+    """
+    Labster License update handlers.
+    """
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminUser,)
+
+    def post(self, request):
+        """
+        Update course license with new simulations list.
+        """
+        license_code = request.data.get('name')
+
+        try:
+            course_license = CourseLicense.objects.get(license_code=license_code)
+        except CourseLicense.DoesNotExist:
+            return Response(
+                {"error": "License %s not found." % license_code},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        ccx_id = course_license.course_id.ccx
+        ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
```

Comment:
Are you sure the course exists?
What if it has been removed? In this case, course_license is still exist.